src/structures/SelectMenuComponent.js

```diff
@@ -0,0 +1,205 @@
+   */
+  setCustomId(customId) {
+    this.customId = Util.verifyString(customId, RangeError, 'SELECT_MENU_CUSTOM_ID');
+    return this;
+  }
+
+  /**
+   * Sets the interactive status of the select menu
+   * @param {boolean} [disabled=true] Whether this select menu should be disabled
+   * @returns {SelectMenuComponent}
+   */
+  setDisabled(disabled = true) {
+    this.disabled = disabled;
+    return this;
+  }
+
+  /**
+   * Sets the maximum number of selections allowed for this select menu
+   * @param {number} maxValues Number of selections to be allowed
+   * @returns {SelectMenuComponent}
+   */
+  setMaxValues(maxValues) {
+    this.maxValues = maxValues;
+    return this;
+  }
+
+  /**
+   * Sets the minimum number of selections required for this select menu
+   * <info>This will default the maxValues to the number of options, unless manually set</info>
+   * @param {number} minValues Number of selections to be required
+   * @returns {SelectMenuComponent}
+   */
+  setMinValues(minValues) {
+    this.minValues = minValues;
+    return this;
+  }
+
+  /**
+   * Sets the placeholder of this select menu
+   * @param {string} placeholder Custom placeholder text to display when nothing is selected
+   * @returns {SelectMenuComponent}
+   */
+  setPlaceholder(placeholder) {
+    this.placeholder = Util.verifyString(placeholder, RangeError, 'SELECT_MENU_PLACEHOLDER');
+    return this;
+  }
+
+  /**
+   * Adds options to the select menu.
+   * @param {...SelectMenuComponentOptionData|SelectMenuComponentOptionData[]} options The options to add
+   * @returns {SelectMenuComponent}
+   */
+  addOptions(...options) {
+    this.options.push(...this.constructor.normalizeOptions(options));
+    return this;
+  }
+
+  /**
+   * Sets the options of the select menu.
+   * @param {...SelectMenuComponentOptionData|SelectMenuComponentOptionData[]} options The options to set
+   * @returns {SelectMenuComponent}
+   */
+  setOptions(...options) {
+    this.spliceOptions(0, this.options.length, options);
+    return this;
+  }
+
+  /**
+   * Removes, replaces, and inserts options in the select menu.
+   * @param {number} index The index to start at
+   * @param {number} deleteCount The number of options to remove
+   * @param {...SelectMenuComponentOptionData|SelectMenuComponentOptionData[]} [options] The replacing option objects
+   * @returns {SelectMenuComponent}
+   */
+  spliceOptions(index, deleteCount, ...options) {
+    this.options.splice(index, deleteCount, ...this.constructor.normalizeOptions(...options));
+    return this;
+  }
+
+  /**
+   * Transforms the select menu into a plain object
+   * @returns {APIMessageSelectMenu} The raw data of this select menu
+   */
+  toJSON() {
+    return {
+      custom_id: this.customId,
+      disabled: this.disabled,
+      placeholder: this.placeholder,
+      min_values: this.minValues,
+      max_values: this.maxValues ?? (this.minValues ? this.options.length : undefined),
```

Comment:
Is the defaulting to this.options.length needed? I don't recall.. 👀

Reply:
I'm assuming so, since it was being done in `MessageSelectMenu`